design/design.md

```diff
@@ -0,0 +1,175 @@
+
+There will be a single script UTxO for the `Pot`, several `PelletState` UTxOs and a `ShipState` UTxO for every user. The `Pot` UTxO locks the ada amount paid by each user when creating a ship, and it's position on the board is always assumed to be (0,0). Both `PelletState` and `ShipState` UTxOs have their positions specified in the datum. In order to identify valid game UTxOs, the admin will deposit a special token in the `PelletState` and `Pot` UTxOs when creating them.
+
+Each ship will be identified by a `ShipToken`, with a fixed policy id but a token name of their own. This is the token that is minted by the _Ship minting policy_, described in the validators section.
+
+
+## UTxOs specification
+
+### ShipState UTxO:
+
+>#### Address
+>* Parameterized on admin token, `Pot` validator address and `PelletState` validator address.
+>
+>#### Datum
+>* pos_x: **Int**
+>* pos_y: **Int**
+>* fuel: **Int**
+>* pilot_token: **ByteArray**
+>
+>#### Value
+>* minAda
+>* `ShipToken`
+
+### PelletState UTxO:
+
+>#### Address
+>* Parameterized on the admin token.
+>
+>#### Datum
+>* pos_x: **Int**
+>* pos_y: **Int**
+>* fuel: **Int**
+>* ship_token_policy: **PolicyId**
+>
+>#### Value
+>* minAda
+>* admin token
+
+### Pot UTxO:
+
+>#### Address
+>* Parameterized on the admin token.
+>
+>#### Datum
+>* ship_token_policy: **PolicyId**
+>
+>#### Value
+>* minAda
+>* reward amount.
+>* admin token
+>
+>Note: the reward amount is updated during the course of the game, whenever a new user joins or reaches the pot.
+
+
+## Transactions
+
+### Create Pot UTxO:
+This transaction creates the unique `Pot` UTxO locking min ada and an admin token.
+
+![createPot diagram](img/createPot.png)
+
+
+### Create a PelletState UTxO:
+Creates one `PelletState` UTxO locking min ada and an admin token, setting in the datum the `pos_x` and `pos_y` coordinates where the pellet will be located on the grid and the `fuel` value equal to some initial value.
+
+![createPellet diagram](img/createPellet.png)
+
+
+### Create a ShipState UTxO:
+Creates a `ShipState` UTxO locking min ada and a ship token (minted in this tx), specifying in the datum the initial `pos_x` and `pos_y` coordinates of the ship, and setting `fuel` to an initial amount. Also adds to the `Pot` UTxO value the inscription fee paid by the user.
+
+![createShip diagram](img/createShip.png)
+
+
+### Move a Ship:
+Updates the `pos_x`, `pos_y` and `fuel` datum fields of the `ShipState` UTxO by adding the `delta_x` and `delta_y` values specified in the redeemer, and subtracting the fuel amount needed for the displacement.
+
+![moveShip diagram](img/moveShip.png)
+
+
+### Gather Fuel:
+Updates the `fuel` datum field of both the `ShipState` and `PelletState` UTxOs, adding the `amount` (specified in the redeemer) from the first and subtracting it from the latter.
+
+![gatherFuel diagram](img/gatherFuel.png)
+
+
+### Collect Rewards:
```

Comment:
User wallet must be shown in the inputs, given that the pilot token is needed for this operation.